Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -6339,6 +6340,181 @@ bool Document::fullscreen_enabled() const
+        // 11. If doc’s fullscreen element is null, then resolve promise with undefined and terminate these steps.
+        if (!doc->fullscreen_element()) {
+            WebIDL::resolve_promise(*realm, promise, JS::js_undefined());
+            return;
+        }
+        // 12. Let exitDocs be the result of collecting documents to unfullscreen given doc.
+        auto exit_docs = doc->collect_documents_to_unfullscreen();
+        // 13. Let descendantDocs be an ordered set consisting of doc’s descendant navigables' active documents whose fullscreen element is non-null, if any, in tree order.
+
+        Vector<GC::Ptr<Document>> descendant_docs;
+        for (auto& descendant : doc->descendant_navigables()) {
+            if (descendant->active_document()->fullscreen_element())
+                descendant_docs.append(descendant->active_document());
+        }
+
+        // 14. For each exitDoc in exitDocs:
+        for (auto const& exit_doc : exit_docs) {
+            // Append (fullscreenchange, exitDoc’s fullscreen element) to exitDoc’s list of pending fullscreen events.
+            exit_doc->append_pending_fullscreen_change(PendingFullscreenEvent::Type::Change, *exit_doc->fullscreen_element());
+            if (resize) {
+                // If resize is true, unfullscreen exitDoc.
+                for (auto el : exit_doc->top_layer_elements()) {
+                    if (el->is_fullscreen_element())
+                        exit_doc->unfullscreen_element(el);
+                }
+            } else {
+                // Otherwise, unfullscreen exitDoc’s fullscreen element.
+                exit_doc->unfullscreen_element(*exit_doc->fullscreen_element());
+            }
+        }
+        // 15. For each descendantDoc in descendantDocs:
+        for (auto& descendant_doc : descendant_docs) {
+            auto el = descendant_doc->fullscreen_element();
+            // Append (fullscreenchange, descendantDoc’s fullscreen element) to descendantDoc’s list of pending fullscreen events.
+            descendant_doc->append_pending_fullscreen_change(PendingFullscreenEvent::Type::Change, *el);
+            // Unfullscreen descendantDoc.
+            Vector<GC::Ref<Element>, 8> fullscreen_elements;
+            for (auto const& element : descendant_doc->top_layer_elements()) {
+                if (element->is_fullscreen_element())
+                    fullscreen_elements.append(element);
+            }
+
+            for (auto& el : fullscreen_elements) {
+                descendant_doc->unfullscreen_element(el);
+            }
+        }
+        // NOTE: The order in which documents are unfullscreened is not observable, because run the fullscreen steps is invoked in tree order.
+        // 16. Resolve promise with undefined.
+        WebIDL::resolve_promise(*realm, promise, JS::js_undefined());
+    }));
+    return promise;
+}
+
+bool Document::is_simple_fullscreen_document() const
+{
+    u32 total = 0;
```

Comment:
why are we counting here? are there spec steps involved? if not, this should just be `bool has_fullscreen_element`, with |= in the loop. `if element_is_fullscreen && has_fullscreen_element: return false` or something. adding bools to ints is icky.

Reply:
We're counting because a simple fullscreen document is one with only 1 element in it's top layer that has it's fullscreen flag set. I'll add a spec link and also just ++ increment.